specs/overview.md

```diff
@@ -0,0 +1,158 @@
+	- A standalone, stateless binary.
+	- Syncs and verifies rollup data on L1.
+	- Applies rollup-specific block production rules to synthesize blocks from L1.
+	- Appends blocks to the L2 chain using the Engine API.
+	- Handles L1 reorgs.
+	- Distributes unsubmitted blocks to other rollup nodes.
+- **Execution Engine (EE)**:
+	- A vanilla Geth node with minor modifications to support Optimism.
+	- Maintains L2 state.
+	- Sync state to other L2 nodes for fast onboarding.
+	- Serves the Engine API to the rollup node.
+- **Batch Submitter**
+	- A background process that submits transaction batches to the `BatchInbox` address.
+- **Output Submitter**
+	- A background process that submits L2 output commitments to the `L2OutputOracle`.
+
+### Transaction/Block Propagation
+
+**Spec links:**
+
+- [Execution Engine](specs/exec-engine.md)
+
+Since the EE uses Geth under the hood, Optimism uses Geth's built-in peer-to-peer network and transaction pool to propagate transactions. The same network can also be used to propagate submitted blocks and support snap-sync. 
+
+Unsubmitted blocks, however, are propagated using a separate peer-to-peer network of Rollup Nodes. This is optional, however, and is provided as a convnience to lower latency for users.
+
+The below diagram illustrates how the sequencer and verifiers fit together:
+
+![Propagation](./assets/propagation.svg)
+
+## Key Interactions In Depth
+
+### Deposits
+
+**Spec links:**
+
+- [Deposits](./deposits.md)
+
+Optimism supports two types of deposits: user deposits, and L1 attributes deposits. To perform a user deposit, users call the `depositTransaction` method on the `DepositFeed` contract. This in turn emits `TransactionDeposited` events, which the rollup node reads during block derivation.
+
+L1 attributes deposits are used to register L1 block attributes (number, timestamp, etc.) on L2 via a call to the L1 Attributes Predeploy. They cannot be initiated by users, and are instead added to L2 blocks automatically by the rollup node.
+
+Both deposit types are represented by a custom EIP-2718 transaction type on L2.
+
+### Block Derivation
+
+#### Overview
+
+The rollup chain can be deterministically derived given an L1 Ethereum chain. The fact that the entire rollup chain can be derived based on L1 blocks is _what makes OE a rollup_. This process can be represented as:
+
+```
+derive_rollup_chain(l1_blockchain) -> rollup_blockchain
+```
+
+Optimism's block derivation function is designed such that it:
+
+- Requires no state other than what is easily accessible using L1 and L2 execution engine APIs.
+- Supports sequencers and sequencer consensus.
+- Is resilient to sequencer censorship.
+
+#### Epochs and the Sequencing Window
+
+The rollup chain is subdivided into epochs. There is a 1:1 correspondence between L1 block numbers and epoch numbers. For L1 block number `n`, there is a corresponding rollup epoch `n` which can only be derived after L1 block number `n` is added to the L1 chain. 
```

Comment:
Not false, but incomplete: epoch `n` can only be fully derived after block `n` + SEQUENCER_WINDOW_SIZE blocks have been added to the L1 chain.